### PR TITLE
feat(audit): auto-generated, add append-only agent audit trail (.beads/interactions.jsonl) + bd audit commands (fixes #649)

### LINKED ISSUE #649: Feature Request: Audit Trail for Agent Interactions
Log agent interactions (prompts, responses, tool calls) to .beads/interactions.jsonl. Allow labeling entries as good or bad after the fact. This enables both auditing ("why did the agent do that?") and fine-tuning dataset generation (so models get better at Beads workflows over time).

---

### Why?

Beads tracks *what changed* (issues, code) but not *how the agent got there*. We lose the prompts, the model outputs, the tool calls, and whether they worked or failed.

An audit trail would let us:

1. **Train better models** — Real task data from real repos makes great SFT/RL training sets
2. **Debug agent behavior** — "Why did it do that?" becomes answerable
3. **Label outcomes** — Mark responses as good/bad after seeing what happened

Since Beads already syncs JSONL to git, the audit log would naturally be versioned and portable.

---

### Current State
Currently: only the LLM client stores the conversation, the prompts/responses are ephemeral and get discarded.

---

### What would it look like?

**A new file:** `.beads/interactions.jsonl`

Each line is one event. Append-only (labels are new entries, not edits).

```jsonl
{"id":"int-a1b2","kind":"llm_call","model":"claude-3-5-haiku","prompt":"Summarize this issue...","response":"**Summary:** Fixed auth bug...","issue_id":"bd-42","created_at":"2024-12-19T10:30:00Z"}
{"id":"int-c3d4","kind":"label","parent_id":"int-a1b2","label":"good","reason":"Accurate summary, merged without changes"}
```

**New CLI commands:**

```bash
# Record an interaction (or pipe JSON via stdin)
bd audit record --kind llm_call --prompt "..." --response "..."

# Label an existing entry
bd audit label int-a1b2 --label good --reason "Worked perfectly"
bd audit label int-c3d4 --label bad --reason "Hallucinated a file path"
```

**Automatic logging:** Built-in LLM features (like `bd compact`) would log their prompts/responses automatically.

---

### Auto-labeling ideas (for RL pipelines)

You don't want to manually label thousands of entries. Here are signals you can extract automatically:

| Signal | Good | Bad |
|--------|------|-----|
| Tool exit code | 0 | non-zero |
| Retries | None needed | Multiple attempts |
| PR outcome | Merged | Closed without merge |
| Reverts | None | Reverted within N days |
| Issue lifecycle | Closed, stays closed | Reopened, spawns fix issues |
| CI results | Green | Red |

A post-processing script reads `interactions.jsonl`, correlates with git history / CI, and appends label entries.

---

### Acceptance criteria

- [ ] `.beads/interactions.jsonl` exists, append-only
- [ ] `bd audit record` works
- [ ] `bd audit label` works  
- [ ] `bd sync` includes `interactions.jsonl`
- [ ] `bd compact` logs prompt/response when enabled

### PR DESCRIPTION
**Pull request by dchichkov**
### PR subject
**Auto-generated: Add append-only agent audit trail (`.beads/interactions.jsonl`) + `bd audit` commands (fixes #649)**

### PR description
**This pull request was automatically generated.**

Implements an append-only audit trail for agent interactions to support auditing and dataset generation. Ref: [Feature Request: Audit Trail for Agent Interactions #649](https://github.com/steveyegge/beads/issues/649).

#### What’s included
- **New file**: `.beads/interactions.jsonl` (append-only; labeling appends new entries)
- **New commands**:
  - `bd audit record` (flags or piped JSON via stdin)
  - `bd audit label <entry-id> --label good|bad --reason "..."`
- **Sync support**: `bd sync` now includes `.beads/interactions.jsonl` in its explicit staging allowlist
- **Compact integration**: `bd compact --auto` logs the LLM prompt/response when enabled (`--audit` or `audit.enabled=true`)
- **Docs**: `commands/audit.md`

#### Notes
- Logging is **best-effort**: audit write failures do not break `bd compact`.
- Hooks were updated to stage/check `interactions.jsonl` alongside other `.beads/*.jsonl` files.

### CODE REVIEW

**Comment by steveyegge** [comment]
## Review Notes

Thanks for the audit trail feature! A couple items before merge:

1. **Test data in PR**: The `.beads/interactions.jsonl` file contains test entries (lines 3-4). This should be empty or the file should not be included in the PR.

2. **CI hasn't run**: Please trigger CI to verify tests pass.

Once those are addressed, this looks good to merge.

**Comment by steveyegge** [comment]
Merged via cherry-pick with fixes (removed test data, added interactions.jsonl to hooks.go for the new thin-shim pattern, resolved merge conflicts). Thank you @dchichkov for this excellent contribution! 🎉

The audit trail feature is now live on main.

**Comment by dchichkov** [comment]
@steveyegge    Cursor-specific hook is available here: 
https://github.com/steveyegge/beads/commit/e82963257cb422f6669e0c16b4068311d846c96e

(it's on https://github.com/steveyegge/beads/compare/main...dchichkov:beads:feature-audit
 branch)
